docs/osf-trolling-list/build_process.md

```diff
@@ -0,0 +1,127 @@
+`[OSFI001]` Build process
+-------------------------
+
+# `[OSFI0010]` There is no healthy OSF without reproducible builds
+
+## What are the reproducible builds?
+
+[Reproducible Builds](https://reproducible-builds.org/) and lot of its aspects
+is well explained on linked website.
+
+## Why those are in particular important for OSF?
+
+This is because knowing hashes of firmware components we can calculate expected
+TPM PCRs values. Knowing good PCR values and being able to predict further
+values after firmware update, we gaining following advantages:
+
+* We can easily confirm if firmware is valid by reading TPM PCRs and compare to
+  reference integrity measures (RIM), RIM can be calculated by build exactly
+  the same version of OSF
+* Process of replying measurements provided in TPM Event Log is simpler because
+  we can easily obtain software and confirm hashes used during PCR extension
+  process were valid and the same as produced by software build process
+* Looking for bugs by using bisection is easier since being able to build every
+  commit of software in reproducible manner give us confidence that there would
+  be no change in behavior between 2 consecutive builds
+* Consistent behavior for given hash can be confirmed by test results tied to
+  given hash in that way users looking at test results may expect exactly the
+  same behavior for firmware binaries with the same hash
+* Long term maintenance should be easier if firmware build would be
+  reproducible over long time, because often happen some firmware land in
+  industrial applications e.g. robots, trains, smart city infrastructure, if
+  bug will be detected after long time it is important to have tools and be
+  able to confirm the same binary can be generated in future. If we know given
+  toolchain reproduce binary bit by bit we are sure it also reproduce
+  software/firmware behavior. In that light making small incremental
+  improvement using toolchain that give reproducible results give us higher
+  chance that we will not introduce uncontrolled change in behavior and code
+  change will be reflected by compilation process without affecting previous
+  software behavior.
+
+# [OSFI0011] Docker containers as build systems "considered harmful"
+
+Docker containers for couple last years become default method for provide
+reproducible runtime environment for software as well those started to be very
+useful for developers to transfer exact configuration for building given
+software stack. We wrote about usefulness of Docker containers in embedded
+software development environment on 
+[3mdeb blog](https://blog.3mdeb.com/2018/2018-09-27-optimize-performance-in-docker/).
+
+After long time of using Docker containers for embedded software development
```

Comment:
Already provided the comments at https://github.com/Dasharo/docs/pull/4/files . In short, I think the reproducible builds are possible with Docker, it's just the resulting Docker images would be bigger and slightly harder to distribute.

Reply:
@mikebdp2 we would need proves of that, I can write something in that lines if 3mdeb in practical use cases will provide it works. At this point, we know maintaining toolchains with Docker seems to be problematic. We will definitely get back to the topics after some experiments would be performed, btw in PR4 I already included what you suggested. Is there any specific action you would like me to do here please let me know.